src/ex3/ItemManager.js

```diff
@@ -0,0 +1,124 @@
+import { PokemonClient } from './PokemonClient.js';
+import { Console } from 'console';
+import { createWriteStream, createReadStream, readFile, writeFile, read } from 'fs';
+import readline from 'readline';
+import { FailedFetchPokemonError, PokemonAlreadyInError} from './Errors.js';
+import chalk from 'chalk';
+import gradient from 'gradient-string';
+
+
+export class ItemManager {
+    constructor() {
+        this.pokemonClient = new PokemonClient();
+        this.typesColors = {
+            "normal": "white",
+            "fighting": "red",
+            "flying": "yellow",
+            "poison": "magenta",
+            "bug": "magenta",
+            "ghost": "magentaBright",
+            "steel": "whiteBright",
+            "fire": "redBright",
+            "water": "blue",
+            "grass": "greenBright",
+            "electric": "yellowBright",
+            "ice": "cyanBright"};
+    }
+
+    addRegTask(task) {
+        myLogger.log(task);
+    };
+
+    colorType(type) {
+        if (this.typesColors[type] !== undefined){
+            return chalk[this.typesColors[type]](type);
+        }
+        switch(type) {
+            case('rock' || 'ground'):
+                return chalk.rgb(205, 127, 50)(type);
+            case('psychic'):
+                return gradient('magenta', 'pink', 'white', 'magenta')(type);
+            case('dragon'):
+                return gradient.pastel(type);
+            case('fairy'):
+                return chalk.rgb(255,192,203)(type);
+        }
+    }
+
+    async addPokemon(pokemon) {
+        const type = this.colorType(pokemon.types[0].type.name);
+        const task = `Catch ${chalk.bold(pokemon.name)}, the ${type} type pokemon`;
+        const items = createReadStream("itemManager.txt", 'utf-8');
```

Comment:
i would separate the whole file operations to a different file name fileManager or something
again, if you'll want to change your app to save the data in some DB instead of a text file you'll change only one file and a very scoped logic

These are critical things to look out for, it becomes harder and harder when your projects will get larger, and a good separation is the absolute key for code readability reusability and easy refactoring later on